src/pages/index.js

```diff
@@ -34,50 +36,126 @@ const enableValidation = (formSettings) => {
+/* ************************* */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authorization: "a0548a22-3cdf-48eb-9822-7e7daf1b0604",
+});
+
 /* ****************************** */
 /* Declaration of UserInfo object */
 /* ****************************** */
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
   jobSelector: ".profile__job",
+  avatarSelector: ".profile__picture",
 });
 
 /* ********************** */
 /* Declarations of popups */
 /* ********************** */
+const avatarForm = new PopupWithForm("form[name='avatar-form']", (data) => {
+  avatarForm.submitButton.textContent = "Saving...";
+  api
+    .editProfilePicture(data.link)
+    .then((res) => (res.ok ? true : res.status))
+    .then(() => {
+      userInfo.setAvatar(data.link);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      avatarForm.submitButton.textContent = "Save";
+      avatarForm.close();
+    });
+});
+avatarForm.setEventListeners();
+
 const cardForm = new PopupWithForm("form[name='card-form']", (data) => {
-  cardGallery.addItem(data);
-  cardForm.close();
+  cardForm.submitButton.textContent = "Saving...";
+  api
+    .addNewCard({ name: data.name, link: data.link })
+    .then((res) => (res.ok ? res.json() : res.status))
+    .then((res) => {
+      cardGallery.addItem(res);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      cardForm.submitButton.textContent = "Create";
+      cardForm.close();
+    });
 });
 cardForm.setEventListeners();
 
 const profileForm = new PopupWithForm("form[name='profile-form']", (data) => {
-  userInfo.setUserInfo(data);
-  profileForm.close();
+  profileForm.submitButton.textContent = "Saving...";
+  api
+    .editUserInfo({ name: data.name, job: data.job })
+    .then((res) => (res.ok ? true : res.status))
+    .then(() => {
+      userInfo.setUserInfo({ name: data.name, about: data.job });
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      profileForm.submitButton.textContent = "Save";
+      profileForm.close();
+    });
 });
 profileForm.setEventListeners();
 
+const confirmDeletePopup = new PopupDeleteConfirmation(
+  "form[name='delete-form']"
+);
+confirmDeletePopup.setEventListeners();
+
 const picturePopup = new PopupWithImage(".popup__picture");
 picturePopup.setEventListeners();
 
 /* *********************************** */
 /* Declaration of card gallery section */
 /* *********************************** */
-const cardGallery = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const card = new Card(data, ".card-template", (data) => {
-        picturePopup.open(data);
+const cardGallery = new Section((data) => {
+  const card = new Card(
+    data,
+    ".card-template",
+    (data) => {
+      picturePopup.open(data);
+    },
+    () => {
+      confirmDeletePopup.setSubmitAction(() => {
+        api.deleteCard(data._id);
+        card.deleteCard();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.